### PR TITLE
Partial doc load

### PR DESCRIPTION
**Pull request by noofherder**
UI only displays a chunk of the doc (if large), with the option to load the whole thing. This improves the UX in the case of very large docs.

### CODE REVIEW

File: src/main/java/com/github/flaxsearch/api/DocumentData.java [hunk 1]
```diff
@@ -22,13 +22,42 @@
 public class DocumentData {
 
     public final ImmutableMultimap<String, String> fields;
+    public final long totalLengthInChars;
+    public final boolean complete;
 
-    public DocumentData(Document document) {
+    public DocumentData() {
```
Comment (romseygeek) [inline src/main/java/com/github/flaxsearch/api/DocumentData.java:28]: Do we need the no-arg constructor?

**Review by romseygeek — COMMENTED** [review]
Looks good.  Do you think we should make the 'load all' box per-field?  ie, what happens if we have two big fields and only want to look at one of them?

File: src/main/java/com/github/flaxsearch/api/DocumentData.java (hunk 1 above)
Comment (noofherder) [reply src/main/java/com/github/flaxsearch/api/DocumentData.java:28]: We need the no-arg constructor for unit tests to run.

**Comment by noofherder** [comment]
I'm not sure about per-field "load all" for the moment. It's more complex to implement and for the user. Shall we wait and see if anyone actually wants it?

**Comment by romseygeek** [comment]
Sounds like a plan.  +1 to merge

**Pull request merged**